MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.h

```diff
@@ -51,17 +53,35 @@ class PiecewiseLinearInterpolation final
     /**
      * \brief Calculates the interpolation value.
      * @param pnt_to_interpolate The point should be located within the range
-     * \f$[x_{\min}, x_{\max}]\f$, where \f$x_{\min} = \min_{1 \le j \le n} x_j\f$ and
-     * \f$x_{\max} = \max_{1 \le j \le n} x_j\f$. Points outside of this interval are
-     * extrapolated.
+     * \f$[x_{\min}, x_{\max}]\f$, where \f$x_{\min} = \min_{1 \le j \le n}
+     * x_j\f$ and
+     * \f$x_{\max} = \max_{1 \le j \le n} x_j\f$. Points outside of this
+     * interval are
+     * set to x_{\min} or x_{\max}.
      * @return The interpolated value.
      */
     double getValue(double pnt_to_interpolate) const;
 
+    /**
+     * \brief Calculates the derivative value.
+     * @param pnt_to_interpolate The point should be located within the range
+     * \f$[x_{\min}, x_{\max}]\f$, where \f$x_{\min} = \min_{1 \le j \le n}
+     * x_j\f$ and
+     * \f$x_{\max} = \max_{1 \le j \le n} x_j\f$. when points are located
+     * outside of this interval
+     * the derivative is set to 0
+     * \attention if the points are located between the first and second points
+     * (or last and second to last point), the derivative is calculated by
+     * simple linear interpolation
+     * otherwise, it is calculated by second order of interpolation with central
+     * difference
```

Comment:
\brief Calculates derivative using quadratic interpolation and central difference quotient.
\attention if the points are located ... , the derivative is calculated using linear interpolation.

Beautyfication option: some lines are broken too early.